src/api/api.ts

```diff
@@ -62,14 +115,168 @@ export class GhostApi {
+      .then((res: ThemeResponse) => {
+        // when the upload of a theme fails the API does not return an error
+        // code but just pass the error as part of the regular response...
+        if (!res.themes) {
+          throw new Error(`${res}`);
+        }
+
+        return res;
+      })
+      .catch(err => {
+        debugLog(`Something went wrong while trying to upload the theme`, err);
+        throw new ErrorActivateTheme();
+      });
+  }
+
+  public uploadRoutes(
+    getReadStream: () => fs.ReadStream
+  ): Promise<RoutesResponse> {
+    if (!this.token) {
+      throw new Error('Before uploading routes you must call the login method');
+    }
+
+    const body = new FormData();
+    body.append('routes', getReadStream());
+
+    return fetch(this.config.urls.uploadRoutesUrl, {
+      method: 'POST',
+      headers: {
+        Authorization: `Bearer ${this.token}`,
+      },
+      body,
+    })
+      .then(res => res.json() as Promise<RoutesResponse>)
+      .then((res: RoutesResponse) => {
+        // when the upload of a theme fails the API does not return an error
+        // code but just pass the error as part of the regular response...
+        if (res.errors) {
+          throw new Error(`${res}`);
+        }
+
+        return res;
+      })
+      .catch(err => {
+        debugLog(`Something went wrong while trying to upload the routes`, err);
+        throw new ErrorUploadRoutes();
+      });
+  }
+
+  public async uploadContent(
+    getReadStream: () => fs.ReadStream
+  ): Promise<ContentResponse> {
+    if (!this.token) {
+      throw new Error(
+        'Before uploading content you must call the login method'
+      );
+    }
+
+    const body = new FormData();
+    body.append('importfile', getReadStream());
+
+    const content = await this.fetchRetryOnErrors(
+      this.config.urls.uploadContentUrl,
+      {
+        method: 'POST',
+        headers: {
+          Authorization: `Bearer ${this.token}`,
+        },
+        body,
+      },
+      ['MaintenanceError']
+    )
+      .then((res: ContentResponse) => {
+        // when the upload of a theme fails the API does not return an error
+        // code but just pass the error as part of the regular response...
+        if (res.errors) {
+          debugLog(
+            `Something went wrong while trying to upload the content`,
+            res.errors
+          );
+          throw new Error(`${res.errors[0].message}`);
+        }
+
+        return res;
+      })
+      .catch(err => {
+        debugLog(
+          `Something went wrong while trying to upload the content`,
+          err.message
+        );
+        throw new ErrorUploadContent(err.message);
+      });
+
+    debugLog('Content uploaded', content);
+
+    return content;
+  }
+
+  private sleep(ms: number): Promise<void> {
+    return new Promise(resolve => setTimeout(resolve, ms));
+  }
+
+  /**
+   * @todo this doesn't work as for some reason the retried fetch times out
```

Comment:
This is a real headscratcher - it seems that ghost fails on first request as a 503, stating that it is still boostrapping, so we retry the fetch after some time, but the second request for some reason _always_ times out...